aioftp/server.py

```diff
@@ -1013,7 +1013,11 @@ def get_paths(self, connection, path):
             else:
                 resolved_virtual_path /= part
         base_path = connection.user.base_path
-        real_path = base_path / resolved_virtual_path.relative_to("/")
+        real_path = base_path / resolved_virtual_path.relative_to(resolved_virtual_path.anchor)
+        try:
+            resolved_virtual_path = pathlib.PurePosixPath("/") / real_path.relative_to(base_path)
+        except ValueError:
+            return base_path, pathlib.PurePosixPath("/")
```

Comment:
```suggestion
        real_path = base_path / resolved_virtual_path.relative_to("/")
        try:
            real_path.relative_to(base_path)
        except ValueError:
            real_path, resolved_virtual_path = base_path, pathlib.PurePosixPath("/")
```

Reply:
1. I chose to replace `"/"` with `.anchor` mostly because in itself it's the usual pattern to remove the anchor. It's not technically necessary here as you pointed out but in my opinion it makes the line in itself more correct (and doesn't really make anything worse either apart from making the line slightly longer)
2. Why remove re-creating the virtual path? I suppose it's just a cosmetic issue since it doesn't affect the `real_path` (which is the important part) in any way but doing it the way I did ensures there's nothing else going on there that wouldn't lead to a traversal outside of `base_path`

Reply:
1. I think `"/"` is more explicit and readable.
2. There is no need to recreate virtual path, since it is created already and can be «valid» or «invalid». Check that real path is in bounds of base path is enough.

Reply:
Both points make sense to me, feel free to go ahead and implement your version, however I can't change it in the PR branch for now since that would require us re-testing the changed behaviour.